client/qiskit_serverless/core/job.py

```diff
@@ -76,13 +77,44 @@ class Configuration:  # pylint: disable=too-many-instance-attributes
     auto_scaling: Optional[bool] = False
 
 
+class JobClient(ABC):
```

Comment:
As we commented through a call `JobClient` I would like to rename it to `JobService` to match it with the current `RunService`.

Apart from that I saw everything good, really good job here @korgan00